source/common/grpc/google_async_client_impl.cc

```diff
@@ -90,6 +90,8 @@ GoogleAsyncClientImpl::GoogleAsyncClientImpl(Event::Dispatcher& dispatcher,
   // have comparable overhead to what we are doing in Grpc::AsyncClientImpl, i.e. no expensive
   // new connection implied.
   std::shared_ptr<grpc::Channel> channel = GoogleGrpcUtils::createChannel(config, api);
+  // Get state with try_to_connect = true to try connection at channel creation.
+  channel->GetState(true);
```

Comment:
Actually the text you wrote as a PR comment response is more complete:
> This cache solution is for initialize gRPC channel at channel creation. This GetState(true) is used to poke the gRPC lb at channel creation, it doesn't have any effect no matter it succeeds or fails. But it helps on initialization. Otherwise, the channel establishment still happens at the first request, same as the behavior of existing code.

Maybe throw this in the comment?

For context, @mattklein123  suggested this could be separately PR'd and tested but I think it might be difficult to test, as it's only a hint to grpc about how to manage its connections. Moreover, this PR doesn't have its desired effect unless this is included, IIUC.

Reply:
Done, added better comments